### PR TITLE
APY calculations

### PR DESCRIPTION
**Pull request by the-emerald**

### CODE REVIEW

File: src/epoch_credits/mod.rs [hunk 1]
```diff
@@ -1 +1,35 @@
...
+
+pub fn get_rewards_for_epoch(epoch: Epoch, client: &RpcClient) -> anyhow::Result<Rewards> {
+    let info = client.get_epoch_info()?;
+
+    // Convert epoch number to slot
+    let start_slot = epoch * info.slots_in_epoch;
+
+    // We cannot use an excessively large range if the epoch just started. There is a chance that
+    // the end slot has not been reached and strange behaviour will occur.
+    // If this is the current epoch and less than 10 slots have elapsed, then do not define an
```
Comment (vkomenda) [inline src/epoch_credits/mod.rs]:
```suggestion
    // If this is the current epoch and less than `SLOT_OFFSET` slots have elapsed, then do not define an
```

File: src/epoch_credits/mod.rs [hunk 2]
```diff
@@ -1 +1,35 @@
+use anyhow::anyhow;
+use solana_client::rpc_client::RpcClient;
+use solana_sdk::clock::Epoch;
+use solana_transaction_status::Rewards;
+
 pub mod caching;
+pub mod caching_metadata;
+
+const SLOT_OFFSET: u64 = 20;
```
Comment (vkomenda) [inline src/epoch_credits/mod.rs]: Not entirely convinced there couldn't be 20 or more skipped slots in a row at the start of an epoch. I think you could try more blocks, possibly in several iterations.

Can you try `SLOT_OFFSET = 1000` or something like that? The RPC call and result wouldn't be too big.

Comment (the-emerald) [reply src/epoch_credits/mod.rs]: I tried `SLOT_OFFSET = 1000` but with `curl` the request took nearly 4 seconds to fulfil using https://api.mainnet-beta.solana.com. Would something like 100 be more appropriate?

Comment (vkomenda) [reply src/epoch_credits/mod.rs]: In that case, sure, 100 or maybe a few requests with smaller intervals would be better.

Comment (the-emerald) [reply src/epoch_credits/mod.rs]: Following up on this - `SLOT_OFFSET` is still 20. Does it need to be changed?

Comment (vkomenda) [reply src/epoch_credits/mod.rs]: Even just doubling it up to 40 would give us plenty of margin. I think 40 would be good.

Comment (vkomenda) [reply src/epoch_credits/mod.rs]: Fixed in 53207a5.

**Review by vkomenda — APPROVED** [review]

**Review by vkomenda — APPROVED** [review]

**Pull request merged**